interpreter.py

```diff
@@ -30,23 +31,47 @@ def main():
     """
     )
     system = ProductSystem(2)
+    program = Program()
+
+    # Read all instructions and set labels
     while True:
         try:
             text = input()
+            if not text:
+                continue
+
+            # Define position of labels
+            tokens = tokenize.LexicalAnalyzer(text).lex()
+            parsed_object = Parser(tokens).parse()
+
+            program.add_instruction(text)
+            program.program_counter += 1
```

Comment:
could incrementing the program counter be the responsibility of `add_instruction`?

Reply:
It certainly could, I'll make the change